pkg/controller/resourceclaim/controller_test.go

```diff
@@ -1127,3 +1131,189 @@ func (em numMetrics) withUpdates(notAllocatedDelta, notAllocatedWithAdminDelta,
+			pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: "test-pod", Namespace: "default"},
+				Spec:       v1.PodSpec{},
+				Status: v1.PodStatus{
+					Phase: v1.PodSucceeded,
+					ExtendedResourceClaimStatus: &v1.PodExtendedResourceClaimStatus{
+						ResourceClaimName: "test-extended-claim",
+					},
+				},
+			},
+			featureGateEnabled:          false,
+			expectEarlyReturn:           true,
+			expectExtendedClaimEnqueued: false,
+		},
+		{
+			name: "deleted pod with extended resource claim",
+			pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: "test-pod", Namespace: "default"},
+				Spec:       v1.PodSpec{},
+				Status: v1.PodStatus{
+					Phase: v1.PodSucceeded,
+					ExtendedResourceClaimStatus: &v1.PodExtendedResourceClaimStatus{
+						ResourceClaimName: "test-extended-claim",
+					},
+				},
+			},
+			featureGateEnabled:          true,
+			expectEarlyReturn:           false,
+			expectExtendedClaimEnqueued: true,
+			deleted:                     true,
+		},
+		{
+			name: "pod with both regular and extended resource claims, completed",
+			pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: "test-pod", Namespace: "default"},
+				Spec: v1.PodSpec{
+					ResourceClaims: []v1.PodResourceClaim{{Name: "regular-claim"}},
+				},
+				Status: v1.PodStatus{
+					Phase: v1.PodSucceeded,
+					ExtendedResourceClaimStatus: &v1.PodExtendedResourceClaimStatus{
+						ResourceClaimName: "test-extended-claim",
+					},
+				},
+			},
+			featureGateEnabled:          true,
+			expectEarlyReturn:           false,
+			expectExtendedClaimEnqueued: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.DRAExtendedResource, test.featureGateEnabled)
+
+			hasResourceClaims := len(test.pod.Spec.ResourceClaims) > 0
+			hasExtendedResourceClaims := utilfeature.DefaultFeatureGate.Enabled(features.DRAExtendedResource) &&
+				test.pod.Status.ExtendedResourceClaimStatus != nil
+			shouldEarlyReturn := !hasResourceClaims && !hasExtendedResourceClaims
+
+			if shouldEarlyReturn != test.expectEarlyReturn {
+				t.Errorf("Expected early return: %v, got: %v", test.expectEarlyReturn, shouldEarlyReturn)
+			}
+
+			if test.expectEarlyReturn {
+				return
+			}
+
+			needsClaims, _ := testPodNeedsClaims(test.pod, test.deleted)
+			shouldProcessClaims := !needsClaims
+
+			if shouldProcessClaims {
+				willProcessExtendedClaim := utilfeature.DefaultFeatureGate.Enabled(features.DRAExtendedResource) &&
+					test.pod.Status.ExtendedResourceClaimStatus != nil
+
+				if willProcessExtendedClaim != test.expectExtendedClaimEnqueued {
+					t.Errorf("Expected extended claim enqueued: %v, got: %v", test.expectExtendedClaimEnqueued, willProcessExtendedClaim)
+				}
+
+				if willProcessExtendedClaim {
+					expectedKey := "claim:" + test.pod.Namespace + "/" + test.pod.Status.ExtendedResourceClaimStatus.ResourceClaimName
```

Comment:
could you check the entry in ec.Queue(), and compare the entry with the expected?